crates/revm/src/cancelled.rs

```diff
@@ -21,3 +23,90 @@ impl Drop for Cancelled {
         self.0.store(true, core::sync::atomic::Ordering::Relaxed);
     }
 }
+
+/// A marker that can be used to cancel execution.
+///
+/// If dropped, it will NOT set the `cancelled` flag to true.
+/// If `cancel` is called, the `cancelled` flag will be set to true.
+///
+/// This is useful in prewarming, when an external signal is received to cancel many prewarming
+/// tasks.
+#[derive(Default, Clone, Debug)]
+pub struct CancelTask(Arc<AtomicBool>);
```

Comment:
should we come up with a different name because task doesn't really fit imo

perhaps `CancellationToken` or `ManualCancel` then we could rename the other one `CancelOnDrop`

Reply:
yeah